### PR TITLE
Add some tests

### PR DESCRIPTION
**Pull request by phoebe-lew**

### CODE REVIEW

**Comment by KendallWeihe** [comment]
I opened a PR into this PR, rather than main https://github.com/TBD54566975/tbdex-kt/pull/3

Could have just yolo merged this... idk just progressing things

**Comment by KendallWeihe** [comment]
So I was thinking... the `.parse()` interface ought to exist on the `Message` and `Resource` abstraction level. Reason being, you could imagine a network interface which is accepting all message types, deserializing ("parsing") them into programmatic types, and then doing conditional logic. Which, this wouldn't be possible as it currently exists because we make the assumption that the calling code can safely assume the given type. 

But, then I reviewed our http spec, and we have dedicated paths for the message types (ex. `POST /exchanges/:id/order`), which means our server code can safely make the assumption, so idk I guess it's fine to parse at the finer-grained abstraction level. 

I think eventually we'll want parse interfaces at all levels, such as...

- `tbdex.parse(JSON)` 
  - `Message.parse(JSON)`
    - `Rfq.parse(JSON)`
    - `Quote.parse(JSON)`
    - ...and so on
  - `Resource.parse(JSON)`
    - `Offering.parse(JSON)`

I'm not actually sure yet... in our ts lib we have it at the `Message.parse()` and `Resource.parse()` level... hmmm

File: protocol/src/test/kotlin/OfferingTest.kt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    offering.sign("fakepk", "fakekid")
+
+    assertThat(offering.signature).isEqualTo("blah")
+  }
+
+  @Test
+  fun `can parse offering from a json string`() {
+    val offering = TestData.getOffering()
+    offering.sign("fakepk", "fakekid")
+    val jsonResource = offering.toString()
```
Comment (KendallWeihe) [inline protocol/src/test/kotlin/OfferingTest.kt:46]: Technically, your "parse" test is also testing the stringification functionality

It's slightly more correct to just hard code a JSON string here in the test

But, it's splitting hairs a bit and we have bigger fish to fry so I like the pragmatism

File: protocol/src/main/kotlin/models/Resource.kt [hunk 2]
```diff
@@ -1,79 +1,67 @@
 package models
 
+import com.fasterxml.jackson.annotation.JsonFormat
+import dateTimeFormat
 import net.pwall.json.schema.JSONSchema
+import typeid.TypeID
 import java.time.OffsetDateTime
```
Comment (ethan-tbd) [inline protocol/src/main/kotlin/models/Resource.kt:7]: did we decide on `OffsetDateTime` because it's better for Java compatibility?

**Review by ethan-tbd — APPROVED** [review]

**Review by mistermoe — APPROVED** [review]
🏅 💯 🍰 

Best

**Pull request merged**